tests/replication/slave/storage/test_events.py

```diff
@@ -348,32 +351,28 @@ def persist(self, backfill=False, **kwargs) -> FrozenEvent:
 
         if backfill:
             self.get_success(
-                self._storage_controllers.persistence.persist_events(
-                    [(event, context)], backfilled=True
-                )
+                self.persistance.persist_events([(event, context)], backfilled=True)
             )
         else:
-            self.get_success(
-                self._storage_controllers.persistence.persist_event(event, context)
-            )
+            self.get_success(self.persistance.persist_event(event, context))
 
         return event
 
     def build_event(
         self,
-        sender=USER_ID,
-        room_id=ROOM_ID,
-        type="m.room.message",
-        key=None,
+        sender: str = USER_ID,
+        room_id: str = ROOM_ID,
+        type: str = "m.room.message",
+        key: Optional[str] = None,
         internal: Optional[dict] = None,
-        depth=None,
+        depth: Optional[int] = None,
         prev_events: Optional[list] = None,
```

Comment:
I think prev_events is probably an `Optional[List[str]]` too.

Reply:
Ahh, I always forget about event v1. Don't worry about this if it's a pain!

Reply:
Yeah it should be `List[Tuple[str, dict]]`. Will push.